libs/net/nodenetworkconfigurationpolicy.py

```diff
@@ -0,0 +1,159 @@
+# API: https://nmstate.io/devel/yaml_api.html#dhcp
+from dataclasses import asdict, dataclass
+from typing import Any
+
+from ocp_resources.exceptions import NNCPConfigurationFailed
+from ocp_resources.node_network_configuration_policy_latest import NodeNetworkConfigurationPolicy
+from ocp_resources.resource import Resource, ResourceEditor
+from timeout_sampler import retry
+
+WAIT_FOR_STATUS_TIMEOUT = 90
+WAIT_FOR_STATUS_SLEEP = 5
+
+
+@dataclass
+class STP:
+    enabled: bool
+
+
+@dataclass
+class BridgeOptions:
+    stp: STP
+
+
+@dataclass
+class Port:
+    name: str
+
+
+@dataclass
+class Bridge:
+    options: BridgeOptions | None = None
+    port: list[Port] | None = None
+    allow_extra_patch_ports: bool | None = None
+
+
+@dataclass
+class IP:
+    dhcp: bool
+    auto_dns: bool = True
+    enabled: bool = False
+
+
+@dataclass
+class IPv4(IP):
+    pass
+
+
+@dataclass
+class IPv6(IP):
+    autoconf: bool = False
+
+
+@dataclass
+class Interface:
+    name: str
+    type: str
+    state: str
+    ipv4: IPv4
+    ipv6: IPv6
+    bridge: Bridge | None = None
+
+
+@dataclass
+class DesiredState:
+    interfaces: list[Interface]
+
+
+class NNCPSuccessStatusNotMet(Exception):
+    pass
+
+
+class NetLibsNodeNetworkConfigurationPolicy(NodeNetworkConfigurationPolicy):
+    """
+    NodeNetworkConfigurationPolicy object.
+    """
+
+    api_group = Resource.ApiGroup.NMSTATE_IO
+
+    class Conditions:
+        class Reason:
+            SUCCESSFULLY_CONFIGURED = "SuccessfullyConfigured"
+            FAILED_TO_CONFIGURE = "FailedToConfigure"
+            NO_MATCHING_NODE = "NoMatchingNode"
+
+    def __init__(
+        self,
+        name: str,
+        desired_state: DesiredState,
+        node_selector: dict[str, str] | None = None,
+    ):
+        """
+        Create and manage NodeNetworkConfigurationPolicy
+
+        Args:
+            name (str): Name of the NodeNetworkConfigurationPolicy object.
+            desired_state (DesiredState): Desired policy configuration - interface creation, modification or removal.
+            node_selector (dict, optional): A node selector that specifies the nodes to apply the node network
+                configuration policy to.
+        """
+        super().__init__(
+            name=name,
+        )
+        self._desired_state = desired_state
+        self._node_selector = node_selector
+
+    @retry(
+        wait_timeout=WAIT_FOR_STATUS_TIMEOUT, sleep=WAIT_FOR_STATUS_SLEEP, exceptions_dict={NNCPSuccessStatusNotMet: []}
+    )
+    def wait_for_status_success(self) -> list[dict]:
```

Comment:
At the moment you are not using the returned value anywhere.
Is there a reason to return a list of dicts?

I do not even understand how this is passing the typing linter. I do not see anything returning a list of dicts in the body.

Reply:
> I'm returning the condition, which is a list of dict.

AFAIK, a condition is a dict, not a list of anything.
The status conditions is a list of dicts.
But now it is gone, so it no longer matters.